### ISSUE TITLE
Keyboard focus order is not logical when navigating through the controls of canvas region.

### ISSUE THREAD

**Issue by msftedad** [xyflow/xyflow#5195]
### What platform were you using when you found the bug?

- React Flow version: 18.3.1
- Browser and version: Edge Version 135.0.3179.66 (Official build) (64-bit)
- OS and version: Windows 11 Version 24H2(OS Build 26120.2415)

### Live code example

https://qhfpxp-5173.csb.app/

### Describe the Bug

On pressing tab key, focus order is incorrect when navigating through the controls of canvas region. Focus is not moving to the 'Button edge 1 in canvas region. It is directly moving to line segments in canvas region.
Current Focus Order: Edges>Edge Controls>Nodes>Zoom in>Zoom out>Fit View>Toggle interactivity>React Flow text

### Steps to reproduce the bug or issue

1. Open the URL: [React Flow Example](https://qhfpxp-5173.csb.app/) in latest edge browser.
2. 'You are opening a CodeSandbox preview, do you want to continue?' page gets displayed.
3. Navigate to 'Yes, Proceed to preview' control using tab key and activate it using enter key.
4. 'React Flow Example' page gets displayed.
5. Press tab key to move to inside Canvas
6. Verify on pressing tab key, focus order is logical when navigating through the controls of canvas region or not.

### Expected behavior

On pressing tab key, focus order should be logical when navigating through the controls of canvas region. Focus should move to the "Button edge 1-->Line segment---close button--Button edge 1 extra.
Expected focus order : React Flow>Button Edge 1(Node)>Connecting line between Button 1 and Button2>Close control>Button Edge 2>Bi direction 1>Upper connecting line Bi direction1 and Bi direction 2>lower connecting line Bi direction1 and Bi direction 2>Edge line if Self connecting node> Connecting node>Zoom in>Zoom out>Fit View>Toggle interactivity

### Screenshots or Videos

https://github.com/user-attachments/assets/837dd0c3-a3a5-409a-a574-824333313d48

![Image](https://github.com/user-attachments/assets/850e9f53-eb73-4332-ad04-a2615a45f012)

### Additional context

NA

**Comment by moklick** [comment]
duplicate of #5189
